src/mds/Beacon.cc

```diff
@@ -173,7 +173,11 @@ void Beacon::send_and_wait(const double duration)
   while (!seq_stamp.empty() && seq_stamp.begin()->first <= awaiting_seq) {
     auto now = clock::now();
     auto s = duration*.95-std::chrono::duration<double>(now-start).count();
-    if (s < 0) break;
+    if (s < 0) {
```

Comment:
I was checking the beacon ack code once again, before writing it's test case, and I think this might be the wrong place to decide that we've a 'missed beacon ack' event. Because this comes before the `cvar.wait_for()` wait call. May be [1] might be the right place be to considered as the event happened? Another method would be, we need to timeout the wait and conclude that we have missed the beacon ack.  @vshankar @lxbsz What's your opinion?

[1] https://github.com/ceph/ceph/blob/master/src/mds/Beacon.cc#L151.